### PR TITLE
Added Timings

### PR DESCRIPTION
**Pull request by TeachMeTW**
Add Timing Instrumentation to Functions

## Overview
Introduces timing instrumentation to functions across the codebase (home, data, db_utils). The aim is to improve performance tracking by measuring the execution time for different stages within each function. By leveraging `ect.Timer()`, we capture detailed timings and store them using `esdsq.store_dashboard_time()` for later analysis.

## Key Changes
- Timings have been added to functions handling data retrieval, processing, and rendering in different components.
- Each function now captures the time taken at crucial steps such as querying data, processing records, updating UI components, and generating plots.
- The total time for each function execution is also recorded, allowing better insights into performance bottlenecks.

## Purpose
These changes provide visibility into the time spent in each stage of function execution, allowing for performance optimization and debugging. The captured timing data can be used for monitoring the efficiency of data-heavy operations, ensuring that the system remains responsive and scalable.

https://github.com/user-attachments/assets/3e9c6309-8652-48a2-b917-2cfddbb6c810

### CODE REVIEW

File: pages/data.py [hunk 1]
```diff
@@ -62,142 +113,262 @@ def update_store_trajectories(start_date: str, end_date: str, tz: str, excluded_
...
 
 @callback(
-    Output('trips-table', 'hidden_columns'), # Output hidden columns in the trips-table
-    Output('button-clicked', 'children'), #updates button label
-    Input('button-clicked', 'n_clicks'), #number of clicks on the button
-    State('button-clicked', 'children') #State representing the current label of button
+    Output('trips-table', 'hidden_columns'),  # Output hidden columns in the trips-table
+    Output('button-clicked', 'children'),  # Updates button label
+    Input('button-clicked', 'n_clicks'),  # Number of clicks on the button
+    State('button-clicked', 'children')  # State representing the current label of button
```
Comment (shankari) [inline pages/data.py:299]: unnecessary whitespace changes.

File: pages/data.py [hunk 2]
```diff
@@ -62,142 +113,262 @@ def update_store_trajectories(start_date: str, end_date: str, tz: str, excluded_
...
+    with ect.Timer() as total_timer:
+
+        # Stage 1: Check if df is a DataFrame and raise PreventUpdate if not
+        with ect.Timer() as stage1_timer:
+            if not isinstance(df, pd.DataFrame):
+                raise PreventUpdate
+        esdsq.store_dashboard_time(
+            "admin/data/populate_datatable/check_dataframe_type",
+            stage1_timer
+        )
```
Comment (shankari) [inline pages/data.py:339]: I am going to merge this now, but I am not sure we need to instrument every single line going forward. This is not doing any data access or any complex computation; it is just an instance check. It should be consistently fast, although it doesn't hurt to check.

File: pages/home.py [hunk 3]
```diff
@@ -133,57 +271,206 @@ def update_card_users(store_uuids):
...
     return fig
 
 
 @callback(
     Output('fig-trips-trend', 'figure'),
     Input('store-trips', 'data'),
-    Input('date-picker', 'start_date'), # these are ISO strings
-    Input('date-picker', 'end_date'), # these are ISO strings
+    Input('date-picker', 'start_date'),  # these are ISO strings
+    Input('date-picker', 'end_date'),  # these are ISO strings
```
Comment (shankari) [inline pages/home.py:430]: More unnecessary whitespace changes

**Review by shankari — APPROVED** [review]
I still see unrelated whitespace changes. Are these being applied automatically by vscode? Can't you turn that off until we check in a separate reformatting PR?

**Comment by TeachMeTW** [comment]
> I still see unrelated whitespace changes. Are these being applied automatically by vscode? Can't you turn that off until we check in a separate reformatting PR?

Yes sorry I apologize; I don't even notice these whitespace changes until they end up in the commit due to the reformatter

**Comment by shankari** [comment]
Can't you turn the reformatter off, given that it causes problems consistently?

**Comment by TeachMeTW** [comment]
I have disabled/removed my autoformatter extension; future commits should be better
